aya/src/maps/sock/sock_hash.rs

```diff
@@ -46,15 +46,15 @@ use crate::{
 /// use aya::maps::SockHash;
 /// use aya::programs::SkMsg;
 ///
-/// let intercept_egress: SockHash<_, u32> = bpf.map("INTERCEPT_EGRESS")?.try_into()?;
+/// let mut intercept_egress = SockHash::<_, u32>::try_from(bpf.map("INTERCEPT_EGRESS")?)?;
```

Comment:
@alessandrod I had to provide explicit types here and I'm not 100% sure why 

My guess is that because we never insert a value into the map with that reference the compiler can't figure out a type for the generic? 

```rust
error[E0282]: type annotations needed for `aya::maps::SockHash<&MapData, V>`
  --> src/maps/sock/sock_hash.rs:49:5
   |
22 | let mut intercept_egress = SockHash::<_,_>::try_from(bpf.map("INTERCEPT_EGRESS")?)?;
   |     ^^^^^^^^^^^^^^^^^^^^
   |
help: consider giving `intercept_egress` an explicit type, where the placeholders `_` are specified
   |
22 | let mut intercept_egress: aya::maps::SockHash<&MapData, V> = SockHash::<_,_>::try_from(bpf.map("INTERCEPT_EGRESS")?)?;
   |                         ++++++++++++++++++++++++++++++++++

error: aborting due to previous error
```

Reply:
Correct, we never `get()` (which is the thing that binds K) so the compiler doesn't know. Explicit type is fine here